src/main/java/org/jabref/model/entry/FileFieldParser.java

```diff
@@ -1,59 +1,58 @@
 package org.jabref.model.entry;
 
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
 
 public class FileFieldParser {
-    public static List<LinkedFile> parse(String value) {
-        if ((value == null) || value.trim().isEmpty()) {
-            return Collections.emptyList();
-        }
 
+    public static List<LinkedFile> parse(String value) {
         List<LinkedFile> files = new ArrayList<>();
-        List<String> entry = new ArrayList<>();
-        StringBuilder sb = new StringBuilder();
-        boolean inXmlChar = false;
-        boolean escaped = false;
 
-        for (int i = 0; i < value.length(); i++) {
-            char c = value.charAt(i);
-            if (!escaped && (c == '\\')) {
-                escaped = true;
-                continue;
-            }
-            // Check if we are entering an XML special character construct such
-            // as "&#44;", because we need to know in order to ignore the semicolon.
-            else if (!escaped && (c == '&') && !inXmlChar) {
-                sb.append(c);
-                if ((value.length() > (i + 1)) && (value.charAt(i + 1) == '#')) {
-                    inXmlChar = true;
+        if ((value != null) && !value.trim().isEmpty()) {
```

Comment:
I like the previous structure more (early return instead of big if), which was also more inline with typical [coding conventions](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement).